### PR TITLE
Add dynamic feature for server side filter

### PR DESCRIPTION
**Pull request by haosong**
- Add dynamic feature for server-side filter
- Update Opentracing SDK version
- Fix corner case for JAX-RS path

### CODE REVIEW

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsDynamicFeature.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  public WavefrontJaxrsDynamicFeature(ApplicationTags applicationTags,
+                                      WavefrontReportingConfig wavefrontReportingConfig) {
+    this.applicationTags = applicationTags;
+    this.wavefrontReportingConfig = wavefrontReportingConfig;
+  }
+
+  @Override
+  public void configure(ResourceInfo resourceInfo, FeatureContext featureContext) {
+    WavefrontReportingConfig wfReportingConfig = this.wavefrontReportingConfig;
```
Comment (sushantdewan123) [inline src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsDynamicFeature.java]: no need of this local variable. 
Can simply use this.wavefrontReportingConfig

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsDynamicFeature.java [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.FeatureContext;
+import javax.ws.rs.ext.Provider;
+
+import io.opentracing.Tracer;
+
+import static com.wavefront.config.ReportingUtils.constructWavefrontSender;
+
+@Provider
+public class WavefrontJaxrsDynamicFeature implements DynamicFeature {
```
Comment (sushantdewan123) [inline src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsDynamicFeature.java:23]: Javadoc?
Also explain how someone can use WavefrontJaxrsDynamicFeature
i.e. what's the use-case?

**Review by sushantdewan123 — APPROVED** [review]
minor comments

File: src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsDynamicFeature.java (hunk 2 above)
Comment (haosong) [reply src/main/java/com/wavefront/sdk/jaxrs/server/WavefrontJaxrsDynamicFeature.java:23]: Basically just adding a few lines of code as below:
```
  @Bean
  public DynamicFeature wavefrontJaxrsFeatureBean(Environment env) {
    ApplicationTags applicationTags =
        constructApplicationTags(env.getProperty("applicationTagsYamlFile"));
    WavefrontReportingConfig wfReportingConfig =
        constructWavefrontReportingConfig(env.getProperty("wfReportingConfigYamlFile"));
    return new WavefrontJaxrsDynamicFeature(applicationTags, wfReportingConfig);
  }
```
Maybe put some example in README.md?

**Pull request merged**